### PR TITLE
Added registration,login and logout

### PR DESCRIPTION
**Pull request by Arishka0904**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2,101 +2,116 @@
...
+            </plugin>
+            <plugin>
+                <groupId>org.flywaydb</groupId>
+                <artifactId>flyway-maven-plugin</artifactId>
+                <version>5.1.4</version>
+            </plugin>
+        </plugins>
+    </build>
 
 </project>
```
Comment (ybotok) [inline pom.xml]: add newline 
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: src/main/java/com/delivery/controller/AdminController.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    private UserServiceImplementation userServiceImplementation;
+    private ProductServiceImplementation productServiceImplementation;
+
+    public AdminController(UserServiceImplementation userServiceImplementation, ProductServiceImplementation productServiceImplementation) {
+        this.userServiceImplementation = userServiceImplementation;
+        this.productServiceImplementation = productServiceImplementation;
+    }
+
+    @GetMapping("/user")
+    public String userList(Model model) {
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/AdminController.java]: plz, use java code naming conventions. Methods must be a verb, not a noun

File: src/main/java/com/delivery/controller/AdminController.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            Map<String, String> errors = ControllerUtils.getErrors(bindingResult);
+            model.mergeAttributes(errors);
+            return "productAdd";
+        }
+
+        productServiceImplementation.addNewProductInDB(product);
+
+        return "redirect:/product";
+    }
+}
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/AdminController.java]: newline =)

File: src/main/java/com/delivery/controller/AdminController.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
+package com.delivery.controller;
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/AdminController.java]: Unfortunately, this class broke the SRP(single responsibility principle). Please, create separate abstractions.

File: src/main/java/com/delivery/controller/AdminController.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    @GetMapping("/product/{product}")
+    public String getProductEdit(@PathVariable Product product, Model model) {
+
+        model.addAttribute("product", product);
+
+        return "productEdit";
+    }
+
+    @PostMapping("/product")
+    public String saveProductEdit(@ModelAttribute ProductDto product) {
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/AdminController.java]: method and class naming should focus on domain logic, not programming.

File: src/main/java/com/delivery/controller/RegistrationController.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            model.addAttribute("messageType", "success");
+            model.addAttribute("message", "User successfully activated");
+        } else {
+            model.addAttribute("messageType", "danger");
+            model.addAttribute("message", "Activation code is not found!");
+        }
+
+        return "login";
+    }
+}
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/RegistrationController.java]: newline =)

File: src/main/java/com/delivery/controller/OrderController.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+public class OrderController {
+    private ProductServiceImplementation productServiceImplementation;
+
+    public OrderController(ProductServiceImplementation productServiceImplementation) {
+        this.productServiceImplementation = productServiceImplementation;
+    }
+
+
+    @GetMapping("/")
+    public String getProductList() {
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/OrderController.java]: wtf?O_o

File: src/main/java/com/delivery/controller/RegistrationController.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    @Value("${recaptcha.secret}")
+    private String secret;
+
+    @GetMapping("/registration")
+    public String registration() {
+        return "registration";
+    }
+
+    @PostMapping("/registration")
+    public String addUser(
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/RegistrationController.java:42]: pls, create a separate validator for this

File: src/main/java/com/delivery/controller/UserController.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    public String updateProfile(
+            @AuthenticationPrincipal User user,
+            @RequestParam String password,
+            @RequestParam String email
+    ){
+        userServiceImplementation.updateProfile(user, password, email);
+
+        return "redirect:/user/profile";
+    }
+}
```
Comment (ybotok) [inline src/main/java/com/delivery/controller/UserController.java]: newline )

File: src/main/java/com/delivery/domain/Role.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+import org.springframework.security.core.GrantedAuthority;
+
+public enum Role implements GrantedAuthority {
+    USER, ADMIN;
+
+    @Override
+    public String getAuthority() {
+        return name();
+    }
+}
```
Comment (ybotok) [inline src/main/java/com/delivery/domain/Role.java]: newline

File: src/main/java/com/delivery/service/MailSender.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        SimpleMailMessage mailMessage = new SimpleMailMessage();
+
+        mailMessage.setFrom(mailProperties.getUsername());
+        mailMessage.setTo(emailTo);
+        mailMessage.setSubject(subject);
+        mailMessage.setText(message);
+
+        mailSender.send(mailMessage);
+    }
+}
```
Comment (ybotok) [inline src/main/java/com/delivery/service/MailSender.java]: newline

**Review by SergeyPentsov — CHANGES_REQUESTED** [review]

**Pull request merged**